oops/backplane/distance.py

```diff
@@ -140,3 +140,276 @@ def center_time(self, event_key):
+        show_info(bp, 'Distance observer to moon center (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+#===============================================================================
+def exercise_sun(bp, printing, saving, dir, refdir,
+                 planet=None, moon=None, ring=None,
+                 undersample=16, use_inventory=False, inventory_border=2):
+    """generic unit tests for distance.py"""
+
+    if planet is not None:
+        test = bp.distance(planet, direction='arr')
+        show_info(bp, 'Distance Sun to planet, arrival (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        test = bp.distance(('sun', planet), direction='dep')
+        show_info(bp, 'Distance Sun to planet, departure (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        test = bp.center_distance(planet, direction='arr')
+        show_info(bp, 'Distance Sun to planet center, arrival (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        test = bp.center_distance(('sun', planet), direction='dep')
+        show_info(bp, 'Distance Sun to planet center, departure (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        ### TODO: use Body.has_rings() once Mark has created it
+        try:
+            test = bp.distance(planet+':ansa', direction='arr')
+            show_info(bp, 'Distance Sun to ansa (km)', test,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+        except:
+            if printing: print("These tests require further development.")
+
+        test = bp.distance(planet+':limb', direction='arr')
+        show_info(bp, 'Distance Sun to limb (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+    if ring is not None:
+        test = bp.distance(ring, direction='arr')
+        show_info(bp, 'Distance Sun to rings, arrival (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        test = bp.distance(('sun', ring), direction='dep')
+        show_info(bp, 'Distance Sun to rings, departure (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        test = bp.center_distance(ring, direction='arr')
+        show_info(bp, 'Distance Sun to ring center, arrival (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+        test = bp.center_distance(('sun', ring), direction='dep')
+        show_info(bp, 'Distance Sun to ring center, departure (km)', test,
+                  printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+#===============================================================================
+def exercise_observer_light_time(bp, printing, saving, dir, refdir,
+                      planet=None, moon=None, ring=None,
```

Comment:
Bad indent

Reply:
Ok, I commented them out and left the warning message.